include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -215,16 +225,20 @@ class IntType : public FIRRTLBaseType, public WidthQualifiedTypeTrait<IntType> {
 public:
   using FIRRTLBaseType::FIRRTLBaseType;
 
-  /// Return an SIntType or UIntType with the specified signedness and width.
+  /// Return an SIntType or UIntType with the specified signedness, width, and
+  /// constness.
   static IntType get(MLIRContext *context, bool isSigned,
-                     int32_t widthOrSentinel = -1);
+                     int32_t widthOrSentinel = -1, bool isConst = false);
 
   bool isSigned() { return isa<SIntType>(); }
   bool isUnsigned() { return isa<UIntType>(); }
 
   /// Return the width of this type, or -1 if it has none specified.
   int32_t getWidthOrSentinel();
 
+  /// Return a 'const' or non-'const' version of this type.
+  IntType getConstType(bool isConst);
```

Comment:
It's probably better to have a getNonConstType flavor rather than a flag.  It doesn't seem like you would be dynamically switching between them.

Reply:
Looking at the [full const cumulative changes](https://github.com/llvm/circt/compare/main...dev/trilorez/firrtl-const-fir?expand=1), in most cases `isConst` is known statically but they are definitely some cases where it is dynamic.

Reply:
Manually switching being `getConstType` and `getNonConstType` in those dynamic cases wouldn't be too big a deal though

Reply:
Actually it does seem pretty useful to have now especially that we no longer guarantee const aggregates have const elements. See [this example](https://github.com/llvm/circt/blob/3e0a0299e8eccec614883e2922b72bfd56e51606/lib/Dialect/FIRRTL/FIRRTLOps.cpp#L2817).

Maybe defaulting `getConstType(bool isConst = true)` and adding `getNonConstType` would be nice.